lib/apiservers/engine/backends/system.go

```diff
@@ -381,3 +408,45 @@ func FetchVolumeStores(client *client.PortLayer) (string, error) {
 
 	return strings.Join(res.Payload.Stores, " "), nil
 }
+
+// RegistrySetContains checks if a URL is within the set of insecure registries
+func RegistrySetContains(registrySet []url.URL, remoteURL *url.URL) bool {
+	containedOK := false
+	remoteIP := net.ParseIP(remoteURL.Hostname())
+	remoteDomains, _ := net.LookupAddr(remoteURL.Hostname())
+
+	for _, rs := range registrySet {
+		// if CIDR was provided for insecure registry range, check if login URL within that ip net
+		if strings.Index(rs.Path, "/") != -1 {
+			_, ipnet, err := net.ParseCIDR(rs.Host + rs.Path)
+			if err == nil && ipnet.Contains(remoteIP) {
+				containedOK = true
+				break
+			}
+		}
+
+		// if wildcard domain was provided for insecure registry range, check if login URL within
+		// that domain
+		if strings.HasPrefix(rs.Host, "*") {
+			rsDomains := strings.SplitAfter(rs.Host, "*.")
+			for _, fqdn := range remoteDomains {
+				if strings.HasSuffix(fqdn, ".") {
+					fqdn = fqdn[:len(fqdn)-1]
+				}
+				domainParts := strings.SplitAfterN(fqdn, ".", 2)
+				if domainParts[1] == rsDomains[1] {
+					containedOK = true
+					break
+				}
+			}
+		}
+
+		// if IP provided for insecure registry
```

Comment:
suggest a comment that we deliberately don't attempt to check FQND<->IP equivalence